wikitextprocessor/core.py

```diff
@@ -1683,6 +1684,7 @@ def magic_repl(m: re.Match) -> str:
         # print("    _finalize_expand:{!r}".format(text))
         return text
 
+    @lru_cache
```

Comment:
Tatu recommends setting a cache size here (the default is surprisingly small, 128, so about 4000 is good?) and that this cache is probably what caused the speed-up because zh.wiktionary has a lot of get_page calls. If * star in the SQL query doesn't contain other fields, then the latter query with explicit titles should be about exactly the same in speed.

Reply:
I think it has more affects on Chinese dump file is because Chinese extractor code call `call_lua_sandbox` less than the English code and `get_page` is the bottleneck. I set the limit to 1000, probably should be enough. The default size already makes `get_page` take less than 1% of the time and reduce the extract time to 2 hours.